### ISSUE TITLE
XmlDomFhirNavigator error on navigating a Bundle.entry.resource

### ISSUE THREAD

**Issue by cknaap** [ewoutkramer/fhir-net-api#628]
If you run the unittest below on the attached xml file, it fails.

```csharp
[BundleWithOneEntry.zip](https://github.com/ewoutkramer/fhir-net-api/files/2145746/BundleWithOneEntry.zip)


using FluentAssertions;
using Hl7.Fhir.Serialization;
using Hl7.FhirPath;
using System.IO;
using System.Linq;
using Xunit;

namespace Fhir.Net.Api.Tests
{
    public class BundleNavigationTests
    {
        [Fact]
        public void XmlBundleEntryNavigationFails()
        {
            var bundleXml = File.ReadAllText("BundleWithOneEntry.xml");
            var xmlNav = XmlDomFhirNavigator.Create(bundleXml);
            var entryNav = xmlNav.Select("entry.resource").First();
            var id = entryNav.Scalar("id");
            id.Should().NotBeNull();
        }
    }
}

```

**Comment by ewoutkramer** [comment]
This bug has been solved in branch "nav-based-serialization", which will be part of the next release!

**Comment by ewoutkramer** [comment]
Fixed by #646.
